pennylane/transforms/metric_tensor.py

```diff
@@ -244,6 +244,14 @@ def circuit(weights):
         This means that in total only the tapes for the first terms of the off block-diagonal
         are required in addition to the circuits for the block diagonal.
     """
+    if not tape.trainable_params:
+        warnings.warn(
+            "Attempted to compute the metric tensor of a tape with no trainable parameters. "
+            "If this is unintended, please mark trainable parameters via the 'requires_grad' "
+            "attribute or 'trainable_params' property."

```

Comment:
One very minor thing but --- unlike `qml.grad` --- `qml.metric_tensor` is used for _all_ autodiff frameworks 🤔 So we might need to make this warning slightly more detailed.

```
"Attempted to compute the metric tensor of a tape with no trainable parameters. "
"If this is unintended, please mark trainable parameters as per your autodiff framework. "
"\n\n    Autograd or PyTorch: set 'requires_grad=True'"
"\n    TensorFlow: Use 'tf.Variable' and call this function inside a gradient tape"
```

Maybe something like that?

Unfortunately there is no workaround for JAX yet, a reason why we should implement #1880 🤔

Reply:
Oh of course, good point 👍